Manual electrode labels + inferred rereference grid

menoua opened the pull request:
<!--
Thanks for contributing a pull request to naplib-python!
-->

#### Reference Issues/PRs
<!--
Example: Fixes #1234. See also #3456.
Please use keywords (e.g., Fixes) to create link to the issues or pull requests
you resolved, so that they will automatically be closed when your pull request
is merged. See https://github.com/blog/1506-closing-issues-via-pull-requests
-->


#### What does this implement/fix? Explain your changes.

Adds option to specify electrode names manually to `process_ieeg`. Also adds option for inferring rereference_grid based on electrode names. Also adds option for filtering electrodes using indices. This filtering is done prior to the previous two features, so it might affect their results.

menoua committed:
Bug fix

menoua committed:
Manual electrode names + inferred rereference grid

menoua committed:
Test for inferred rereference grid

menoua committed:
Fix test

gavinmischler commented on naplib/naplab/process_ieeg.py:
```diff
@@ -65,7 +66,12 @@ def process_ieeg(
+        contact. In both cases, the number of labels provided should match the number of data channels in ``data_path``. If
+        None, the labels included in the data file will be used.
+    rereference_grid : Optional[Union[np.ndarray, str]], default=None
         If not None, then data are re-referenced based on this referencing scheme. If a numpy array, then
```
update this description to include the option for "subject" to be given as a string

menoua committed:
Bug fix

menoua committed:
Ensure labels_data is a numpy array

menoua committed:
Allow filtering electrodes using indices

This feature is most useful in discarding unconnected electrodes included in the recording file.

gavinmischler commented on naplib/io/fileio.py:
```diff
@@ -216,6 +216,9 @@ def load(filename):
     with open(filename, 'rb') as inp:
         output = pickle.load(inp)
 
+    if 'labels_data' in output:
+        output['labels_data'] = np.array(output['labels_data'])
+
```
I don't think this should be added here. This is just a generic load function which can load from anything you pickled. But this assumes the thing you loaded is a dict

menoua replied:
Good point. I moved it to `process_ieeg`.

menoua committed:
Make electrode name sequence a numpy array
